core/src/main/scala/org/broadinstitute/dsde/rawls/monitor/workspace/WorkspaceResourceMonitor.scala

```diff
@@ -66,43 +68,51 @@ class WorkspaceMonitorRouter(val config: WorkspaceManagerResourceMonitorConfig,
 
 }
 
-class WorkspaceResourceMonitor(
+case class WorkspaceResourceMonitor(
   jobDao: WorkspaceManagerResourceMonitorRecordDao,
-  jobRunners: List[WorkspaceManagerResourceJobRunner]
+  jobRunners: Map[JobType, WorkspaceManagerResourceJobRunner]
 )(implicit val executionContext: ExecutionContext)
     extends LazyLogging {
 
-  val registeredRunners: Map[JobType, List[WorkspaceManagerResourceJobRunner]] = jobRunners.groupBy(_.jobType)
-
   /**
     * Run all jobs, and wait for them to complete
+    *
     * @return the message containing the number of uncompleted jobs
     */
   def checkJobs(): Future[CheckDone] = for {
     jobs: Seq[WorkspaceManagerResourceMonitorRecord] <- jobDao.selectAll()
     jobResults <- Future.sequence(jobs.map(runJob))
-  } yield CheckDone(jobResults.count(_ == false))
+  } yield CheckDone(jobResults.count(!_.isDone))
 
   /**
     * Runs the job in all job runners configured for the type of the job
     * Deletes the job if all runners have completed successfully
+    *
     * @return true if all runners for the job completed successfully, false if any failed
     */
-  def runJob(job: WorkspaceManagerResourceMonitorRecord): Future[Boolean] = for {
-    jobResults <- Future.sequence(
-      registeredRunners
-        .getOrElse(job.jobType, List())
-        .map(_.run(job).map(_.complete).recover { case t =>
-          logger.error(s"Exception monitoring WSM Job", t)
-          false
-        })
-    )
-  } yield
-    if (jobResults.forall(a => a)) { // true iff all runners for the given job completed successfully
-      jobDao.delete(job) // remove completed jobs
-      true
-    } else {
-      false
-    }
+  def runJob(job: WorkspaceManagerResourceMonitorRecord): Future[JobStatus] =
+    for {
+      status <- jobRunners.getOrElse(job.jobType, NoopJobRunner)(job).recover { case t: Throwable =>
+        logger.warn(
+          "Failure monitoring WSM job " +
+            s"[ jobId = ${job.jobControlId}" +
+            s", jobType = ${job.jobType}" +
+            s"]",
+          t
+        )
+        Incomplete
+      }
+
+      _ <- Applicative[Future].whenA(status.isDone) {
+        jobDao.delete(job)
+      }
+
+    } yield status
+}
 
+case object NoopJobRunner extends WorkspaceManagerResourceJobRunner {
+  override def apply(job: WorkspaceManagerResourceMonitorRecord)(implicit
+    executionContext: ExecutionContext
+  ): Future[JobStatus] =
+    Future.successful(Incomplete)
```

Comment:
What is the status Incomplete?

Reply:
Oh, I think this is if the job type doesn't match a runner (after seeing the test "mark any jobs that doesnt have a registered handler as incomplete"). I would suggest a comment or rename the JobRunner as something like `NoRegisteredJobRunner`.

Reply:
This is a continutation from Blakes last PR. In his runner implementation, there was always a 1-to-1 relationship between runners and job types. When a runner runs a job, it returns a Status that's either `Incomplete` or `Complete` that's used to schedule the next run. This 1-to-1 relationship is not the case and to make the simplest change possible without redesigning everying, I just returned incomplete for the job types we don't have runners for. I didn't want to make this PR explode more than it needed to. If we did https://broadworkbench.atlassian.net/browse/WOR-533 before this, I wouldn't need to do this.

Reply:
I prefer to name it in terms of what it does to noop might mean that it ends in a completed state. Perhaps `AlwaysIncompleteJobRunner`?